### PR TITLE
Add fastlane for automatic releasing

### PR DESCRIPTION
**Pull request by BohdanBilonoh**
@aol-public/mobile-sdk-team: Ready for review.

[JIRA Issue](https://jira.ops.aol.com/browse/OMSDK-XXXX)

Proposed to automatically create VideoRenderer release using Fastlane.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+desc "Generate VideoRenderer release description"
+lane :release_videorenderer_description do 
+	description = "Template description"
+	begin
+		description = github_changelog_generator(output: "unreleased.md", unreleased_only: true)
+		UI.message description
+		description
+	rescue => exception
+		UI.user_error! exception.message
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: Just return empty description - no reason to treat this as an error.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	end
+end
+
+desc "Check if VideoRenderer version exist"
+lane :is_release_exist do |options|
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: Looks like a separate action that could be reused.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	draft_release = github_releases_api(
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: We don't have any `draft_release` concept for videorenderer.

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	UI.user_error!("Expecting `next_version`!") unless version != nil
+
+	repo_name = "aol-public/OneMobileSDK-videorenderer-ios"
+
+	UI.user_error!("Release on tag #{version} already exists!") unless !is_release_exist(repo_name: repo_name, version: version)
+
+	swift_version = options[:swift_version]
+	UI.user_error! "Expecting `swift_version`!" unless swift_version != nil
+
+	travis_branch = options[:branch]
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: What is the purpose of this check - this will be only triggered from the developers machine.

File: fastlane/Fastfile [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	is_pull_request = options[:is_pull_request]
+
+	is_travis = travis_branch != nil && is_pull_request != nil
+	if is_travis == true
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: The same as above.

File: fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+			framewokr_link_url: asset['browser_download_url'],
+			podspec_path: podspec_path,
+			version: version)
+		deploy_podspec(
+			podspec_path: podspec_path, 
+			spec_repo_name: "onemobile-sdk-podspecs",
+			spec_repo_url: "git@github.com:vidible/OneMobileSDK-releases.git",
+			sources_repo: sources_repo,
+			swift_version: swift_version)
+		reset_git_repo(
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: What is the reason of `reset_git_repo` on each asset?

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

**Comment by AndriiMoskvin** [comment]
Bad name for a PR. No description.
Avoid copy-pasting when possible - you have added a lot of code which is useless for this task.
Please do a cleanup. Having less code - is better.

File: fastlane/Fastfile (hunk 6 above)
Comment (BohdanBilonoh) [reply fastlane/Fastfile]: Not for each, only for `next unless asset['name'] == archive_name`

Comment (AndriiMoskvin) [reply fastlane/Fastfile]: No reason to have this code inside loop :)

File: fastlane/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+fastlane documentation
```
Comment (AndriiMoskvin) [inline fastlane/README.md]: Add this file to gitignore - it is autogenerated.

File: fastlane/Fastfile [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+# Usage like these expected 'fastlane release_videorenderer next_version:1.18 swift_version:3.2 token: 123'
+desc "Release VideoRenderer version"
+lane :release_videorenderer do |options|
+	next if !utils(action: "is_branch_up_to_date", git_branch: git_branch)
+	version = options[:next_version]
+	UI.user_error!("Expecting `next_version`!") unless version != nil
+
+	repo_name = "aol-public/OneMobileSDK-videorenderer-ios"
+
+	UI.user_error!("Release on tag #{version} already exists!") unless github_releases_api(action: "is_release_exist", repo_name: repo_name, version: version, token: options[:token])
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: Format this code properly - it is not readable.

File: fastlane/Fastfile [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+	swift_version = options[:swift_version]
+	UI.user_error! "Expecting `swift_version`!" unless swift_version != nil
+
+	next if !utils(action: "is_branch_up_to_date")
+
+	UI.message "Starting prerelease of VideoRenderer..."
+
+	framework_name = "VideoRenderer"
+	archive_name = framework_name+".framework.zip"
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: Formatting + use template strings.

File: fastlane/Fastfile [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+	carthage(
+		command: "archive",
+		project_directory: "VideoRenderer/",
+		frameworks: framework_name,
+		output: archive_name)
+
+	sources_repo = "https://github.com/vidible/OneMobileSDK-releases.git"
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: This variable is used once - it should be inlined.

File: fastlane/Fastfile [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+	podspec_name = "VideoRenderer.podspec"
```
Comment (AndriiMoskvin) [inline fastlane/Fastfile]: Move this variable closer to usage.

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**